pyccel/codegen/printing/ccode.py

```diff
@@ -953,8 +954,11 @@ def _print_FunctionDef(self, expr):
             self._additional_args.append(expr.results)
         body  = self._print(expr.body)
         decs  = [Declare(i.dtype, i) if isinstance(i, Variable) else FuncAddressDeclare(i) for i in expr.local_vars]
-        if len(expr.results) <= 1 :
-            decs += [Declare(i.dtype, i) if isinstance(i, Variable) else FuncAddressDeclare(i) for i in expr.results]
+        if len(expr.results) == 1 :
+            if isinstance(expr.results[0], Variable) and expr.results[0] != self._unused_variable:
+                decs += [Declare(expr.results[0].dtype, expr.results[0])]
+            elif isinstance(expr.results[0], FunctionAddress):
+                decs += [FuncAddressDeclare(expr.results[0])]
         decs += [Declare(i.dtype, i) for i in self._additional_declare]
```

Comment:
See above. Could it be possible one day for something in `_additional_declare` to also be in `_unused_variable`?

Reply:
did mean that one day `_additional_declare` could have an unused variable?